src/main/java/com/example/ecommerce/controller/ProductController.java

```diff
@@ -0,0 +1,61 @@
+package com.example.ecommerce.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import com.example.ecommerce.dto.CategoryDto;
+import com.example.ecommerce.model.Category;
+import com.example.ecommerce.model.Product;
+import com.example.ecommerce.service.CategoryService;
+import com.example.ecommerce.service.ProductService;
+
+@Controller
+public class ProductController {
+
+    @Autowired
+    private ProductService productService;
+
+    @Autowired
+    private CategoryService categoryService;
+
+    @GetMapping("/products")
+    public String products(Model model){
+        List<CategoryDto> categoryDtoList = categoryService.getCategoryAndProduct();
+        List<Product> products = productService.getAllProducts();
+        List<Product> listViewProducts = productService.listViewProducts();
+        model.addAttribute("categories", categoryDtoList);
+        model.addAttribute("viewProducts", listViewProducts);
+        model.addAttribute("products", products);
+        return "shop";
+    }
+
+    @GetMapping("/find-product/{id}")
+    public String findProductById(@PathVariable("id") Long id, Model model){
+        Product product = productService.getProductById(id);
```

Comment:
bổ sung code nếu ko thấy product thì hiển thị page home cùng mess "đã có lỗi xảy ra, vui lòng load lại page"
hoặc di chuyển đến 404 not found chẳng hạn